### DISCUSSION TITLE
Cannot assign [Post] to IdentifiedArrayOf<Post>

### DISCUSSION THREAD

**Discussion by dadixon** [pointfreeco/swift-identified-collections#34] [Q&A] [answered]
I'm sure there's a quick and easy way around this but if I set my state variable to [] it works but when I put values in there I get: Cannot assign value of type '[Post]' to type 'IdentifiedArrayOf<Post>' (aka 'IdentifiedArray<UUID, Post>')

**Comment by stephencelis** [comment]
@dadixon It looks like you're encountering a type mismatch, where you have a plain `Array` of posts (`[Post]` is sugar for `Array<Post>`) rather than an `IdentifiedArrayOf<Post>`. If you provide more code we could better help point out the problem.

**Comment by dadixon** [comment, marked as answer]
`struct PostsState: Equatable {
    var posts: IdentifiedArrayOf<PostState> = []
}

switch action {

case let getPosts(posts):
    state.posts = posts
}

let posts = [
    PostState(id, other values),
    PostState(id, other values),
    PostState(id, other values)
}`

**Reply by stephencelis** [reply]
@dadixon Swift will infer that `posts` is `Array<Post>` since there's no type annotation.

If you change it to the following, it should work:

```swift
let posts: IdentifiedArrayOf<Post> = [
  ...
]
```

**Reply by dadixon** [reply]
This works. Thanks
